data/test/scenarios/filter_vision.cfg

```diff
@@ -148,22 +106,162 @@
+# That side 4 sees any side.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_all_side_4" (
+    [event]
+        name=start
+
+        {ASSERT_TEST_TRUE (side=4) (side=1,2,3,4)}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [filter_vision],[filter_location][filter_vision]
+##
+# Expected end state:
+# That side 1 sees side 4.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_side_1_side_4" (
+    [event]
+        name=start
+
+        {ASSERT_TEST_TRUE (side=1) (side=4)}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [filter_vision],[filter_location][filter_vision]
+##
+# Expected end state:
+# That side 2 sees side 4.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_side_2_side_4" (
+    [event]
+        name=start
+
+        {ASSERT_TEST_TRUE (side=2) (side=4)}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [filter_vision],[filter_location][filter_vision]
+##
+# Expected end state:
+# That side 3 sees side 4.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_side_3_side_4" (
+    [event]
+        name=start
+
+        {ASSERT_TEST_TRUE (side=3) (side=4)}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [filter_vision],[filter_location][filter_vision]
+##
+# Expected end state:
+# That side side 4 sees side 1.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_side_4_side_1" (
+    [event]
+        name=start
+
+        {ASSERT_TEST_TRUE (side=4) (side=1)}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [filter_vision],[filter_location][filter_vision]
+##
+# Expected end state:
+# That side 4 sees side 3.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_side_4_side_3" (
+    [event]
+        name=start
+
+        {ASSERT_TEST_TRUE (side=4) (side=3)}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [filter_vision],[filter_location][filter_vision]
+##
+# Expected end state:
+# That side 4 does not see side 2.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_side_4_not_side_2" (
+    [event]
+        name=start
+
+        {ASSERT_TEST_FALSE (side=4) (side=2)}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [filter_vision],[filter_location][filter_vision]
+##
+# Expected end state:
+# That no side sees a non-existent side.
+#####
+
+{TEST_FILTER_VISION_SCEN "filter_vision_not_side_5" (
     [event]
         name=start
-        {VARIABLE chat_ctr 0}
-
-        {assert_test_true () (side=1,2,3,4)}
-        {assert_test_true (side=2) (side=1,2,3,4)}
-        {assert_test_true (side=4) (side=1,2,3,4)}
-        {assert_test_true (side=1) (side=4)}
-        {assert_test_true (side=2) (side=4)}
-        {assert_test_true (side=3) (side=4)}
-        {assert_test_true (side=4) (side=1)}
-        {assert_test_true (side=4) (side=3)}
-        {assert_test_false (side=4) (side=2)}
-        {assert_test_false () (side=5)}
 
+        {ASSERT_TEST_FALSE () (side=5)}
         {SUCCEED}
     [/event]
 )}
```

Comment:
Please `undefine` macros that are only for use in this file.

Reply:
Updated.